zio-kafka/src/main/scala/zio/kafka/producer/Producer.scala

```diff
@@ -187,6 +186,10 @@ trait Producer {
 }
 
 object Producer {
+  case object SendOmittedDueToPreviousRecordSendCallFailureError
+      extends RuntimeException("Send omitted due to the previous record send call failure")
+      with NoStackTrace
```

Comment:
In ZIO context we always talk about 'error' not 'failure' (an error is if something goes wrong,  failure is if something bad happens). As a library, we do not know if not being able to send a record is a failure or not. `FailureError` is confusing, we can stick to the java convention: Exception.

I really don't like the very long name, we could move explanation to the message. WDYT?

The exception message can be a bit more elaborate.

I propose:
```suggestion
  case object SendOmittedException
      extends RuntimeException("Send omitted due to a send error for a previous record in the chunk")
      with NoStackTrace
```

Or:
```suggestion
  case object SendOmittedException
      extends RuntimeException("Send omitted due to a send error for an earlier record in the chunk")
      with NoStackTrace
```

I am not a native speaker, a recommendation on what sounds better is appreciated.

Reply:
I like the shorter name, thanks!
Used `Send omitted due to a send error for a previous record in the chunk` message option, 'cause `previous` seems more accurate than `earlier` to me, but Im also not a native speaker.